common/config/azure-pipelines/templates/core-build.yaml

```diff
@@ -4,6 +4,12 @@ parameters:
   - name: workingDir
     type: string
     default: $(System.DefaultWorkingDirectory)
+  - name: rushBuildCacheEnabled
+    type: number
+    default: 1
```

Comment:
do you mean to enable build cache by default? if so does it make more sense to enable it by default in the rush build-config.json, and dont provide a sas key for the container, so we can continue to test this for a few more days locally before enabling it for all our devs?

Reply:
Yeah that sounds good. 

This was quickly reworked to test the build variable logic/syntax - but the intent was to disable for local dev (hence disabled in the config json) and then work on the logic in the pipeline depending on the build situation.

Reply:
Looking for a role in azure for all Bentley users - do you know if that exists? Have never set up a container for public write access before.

Reply:
Does "Contributor" suit the needs ? I think everyone that "uses" Azure is added to this group so we get at least read access to most of the repos.
I dont think it's a role though, probably just a group.

Reply:
Actually, [looking at the relevant rush code](https://github.com/microsoft/rushstack/blob/c7aef037ee57dbb0892b2882c9d8404cc51ce0fc/rush-plugins/rush-azure-storage-build-cache-plugin/src/AzureStorageBuildCacheProvider.ts#L193), I am not sure we can allow anon write access:

  ```ts
 let blobServiceClient: BlobServiceClient;
    if (sasString) {
      const connectionString: string = this._getConnectionString(sasString);
      blobServiceClient = BlobServiceClient.fromConnectionString(connectionString);
    } else if (!this._isCacheWriteAllowedByConfiguration) {
      // If cache write isn't allowed and we don't have a credential, assume the blob supports anonymous read
      blobServiceClient = new BlobServiceClient(this._storageAccountUrl);
    } else {
      throw new Error(
        "An Azure Storage SAS credential hasn't been provided, or has expired. " +
          `Update the credentials by running "rush ${RushConstants.updateCloudCredentialsCommandName}", ` +
          `or provide a SAS in the ` +
          `${EnvironmentVariableNames.RUSH_BUILD_CACHE_CREDENTIAL} environment variable`
      );
    }
```

Reply:
i was hoping if we enabled rush build cache by default and didnt provide a sas token it would just be a no-op but if it throws an error then lets not enable it by default

i was being lazy and hoping to not need to set another env var to test it out along with the sas key, but its nbd and doesnt save us much for testing atm

Reply:
so i think the default here should be 0, if were not enabling build cache by default, https://rushjs.io/pages/configs/environment_vars/#rush_build_cache_write_allowed-experimental

Reply:
Right - thanks

Reply:
so ben and i spoke on teams, i got this mixed up. we do want to enable it in the ci, just dont want to enable it for other users (locally) yet

so we're gonna flip this back to 1, and as long as build cache is not enable for everyone (set in build-cache.json) we should be ok